change-is-inevitable-versioning-eds/README.md

```diff
@@ -0,0 +1,7 @@
+# Change is inevitable: versioning event-driven systems
+
+## Abstract
+
+Building an event-driven system is anything but trivial. Still, once you make it past the eventual consistency hurdles, the sea of pub-sub vs. command-response debates, and the service boundaries conundrum, you’ll soon face a new, inevitable adversary: change. The conversations that follow sound all too familiar: “Who’s subscribed to this message?” “Do other services depend on this field in the payload?” “Why on earth is that thing in the payload?” “That service should never rely on this data!” And, of course, the obvious “Can’t we -just- remove this?”
+
+Change in event-driven systems raises a multitude of questions. How do you measure the impact of tweaking a message contract without forcing other teams beyond their deadlines? Can you make the change without breaking half of the system? Which techniques provide backward compatibility, and, how long should it be maintained? Oh, and let’s not forget that we’re supposed to solve this problem with zero downtime as our users are spread across every time zone. In this session, we’ll discuss practical techniques that can help us deal with change effectively. From versioning techniques to strategies for backward compatibility and tooling that can support us, we’ll explore how to evolve your event-driven system without causing havoc.
```

Comment:
Changes in event-driven systems raise essential questions. **As software developers, we aim to be agents of change, not chaos.** How do you determine the impact of changing a message contract while considering another team's deadlines? Can you make modifications without breaking half of the system? Is backward compatibility critical, and how long should you maintain that compatibility? Remember that we're also expected to solve challenges with zero downtime as users across several time zones access the system. When working with event-driven systems, we don't just want to survive; we want to thrive.

This session will discuss practical techniques for dealing effectively with change. From versioning techniques to strategies for backward compatibility and tooling that can support us, we'll explore how to evolve your event-driven system without causing havoc. After I share these techniques, you'll increase your capacity to deliver change effectively while giving your stakeholders more confidence in the systems you build.

Reply:
> As software developers, we aim to be agents of change, not chaos

Yasss 🔥